File: M1/azor-chatdog-golang/commands/session_list.go
package commands

import (
	"azor-chatdog/cli"
	"azor-chatdog/files"
	"fmt"
)

// ListSessionsCommand lists all available sessions
func ListSessionsCommand() {
	sessions := files.ListSessions()

	if len(sessions) == 0 {
		cli.PrintInfo("Brak zapisanych sesji.")
		return
	}

	cli.PrintInfo("\n=== Dostępne Sesje ===")
	for _, session := range sessions {
		if session.Error != "" {
			cli.PrintError(fmt.Sprintf("  %s - %s", session.ID, session.Error))
		} else {
			cli.PrintInfo(fmt.Sprintf("  %s - %d wiadomości (ostatnia: %s)", session.ID, session.MessagesCount, session.LastActivity))
		}
	}
	cli.PrintInfo("======================\n")
}

File: M1/azor-chatdog-py/src/commands/session_list.py
from files import session_files
from cli import console

def list_sessions_command():
    """Displays a formatted list of available sessions."""
    sessions = session_files.list_sessions()
    if sessions:
        console.print_help("\n--- Dostępne zapisane sesje ---")
        for session in sessions:
            if session.get('error'):
                console.print_error(f"- ID: {session['id']} ({session['error']})")
            else:
                title = session['title'] if session.get('title') else '(brak tytułu)'
                console.print_help(f"- {title} (Wiadomości: {session['messages_count']}, Ost. aktywność: {session['last_activity']}) | {session['id']}")
        console.print_help("-------------------------------")
    else:
        console.print_help("\nBrak zapisanych sesji.")
